vllm/core/block/naive_block.py

```diff
@@ -129,7 +131,7 @@ def _allocate_block_id(self) -> BlockId:
         if not self._free_block_indices:
             raise BlockAllocator.NoFreeBlocksError()
 
-        block_id = self._free_block_indices.popleft()
+        block_id = heapq.heappop(self._free_block_indices)
```

Comment:
Should it be affecting cuda/rocm devices? If not, please add is_hpu check

Reply:
It will not, we tested on cuda device, performance stays the same. This PR will mainly be benefit to ContiguousPA

Reply:
Hi @michalkuligowski , thank you for your comments.

My understanding is that the vllm maintainer might prefer to keep this part of code device agnostic. Since the code changes work for all devices, instead of adding `is_hpu` check, maybe it's better to add an option, such as `EngineArgs.naive_block_allocator_use_heap` to control the behavior. (if that is what you are suggesting).

That being said, in my opinion, the final solution would depends on the community reviewer's preference when upstreaming, so do you think it makes sense that we keep the code as is for now and do necessary changes when we upstream according to the vllm maintainer's request?

Reply:
@kzawora-intel what do you think?